### PR TITLE
[aggregator] Process campaign state without waiting for first campaign check interval

### PR DESCRIPTION
**Pull request by vdarulis**
This fixes an annoying behavior, where campaign manager will always have a delay of $campaignCheckInterval, even if the process has crashed/restarted and lease hasn't expired yet. It also makes other tests that rely on non-mocked-out campaign manager much slower than they need to be.

The initialization delay also exposed some racy behavior in campaign manager tests. Fixing everything and reducing sleeps sped them up by quite a bit:
```
~# go test  github.com/m3db/m3/src/aggregator/aggregator  -tags integration  -count 1
ok  	github.com/m3db/m3/src/aggregator/aggregator	12.460s
```
vs
```
ok  	github.com/m3db/m3/src/aggregator/aggregator	34.935s
```

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/aggregator/aggregator/election_mgr.go [hunk 1]
```diff
@@ -579,19 +579,28 @@ func (mgr *electionManager) checkCampaignStateLoop() {
 	defer ticker.Stop()
 
 	for {
+		select {
```
Comment (abliqo) [inline src/aggregator/aggregator/election_mgr.go:582]: This wait for `doneCh` is redundant as it seems to only cover the case when the manager is closed immediately after opening.

Comment (vdarulis) [reply src/aggregator/aggregator/election_mgr.go:582]: this exact case is hit in election manager tests, where the wait between start/resign/close is minimal - if it does happen, things can deadlock, as default `campaignIsEnabled` function expects placement manager not to block.

Comment (abliqo) [reply src/aggregator/aggregator/election_mgr.go:582]: yeah, I just ran the tests without it and apparently I was wrong. Thanks for explaining!

**Comment by abliqo** [comment]
LGTM

**Review by abliqo — APPROVED** [review]

**Pull request merged**